core/leader/leader.go

```diff
@@ -69,6 +69,12 @@ func Run(db *sql.DB, addr string, lead func(context.Context)) {
 // is leader periodically. Every time the process becomes leader
 // or is demoted from being a leader, it sends a bool on the
 // returned channel.
+//
+// It provides the invariants:
+// * the first value sent on the channel is `true` when the
```

Comment:
Sorry to nitpick, but this could be interpreted as "the first value sent on the channel is true but only under certain conditions: when the process ...".

How about "The first value sent on the channel is true. (This will happen at the time the process is first elected leader.)"?